doc/sphinx/advanced_methods.rst

```diff
@@ -1551,3 +1551,262 @@ bonded particles.
+
+.. .. note:: Requires support for energy calculations for all used interactions since it uses Monte-Carlo moves which use energies.
+
+In the constant pH method due to Reed and Reed
+:cite:`reed92a` it is possible to set the chemical potential
+of :math:`H^{+}` ions, assuming that the simulated system is coupled to an
+infinite reservoir. This value is the used to simulate dissociation
+equilibrium of acids and bases. Under certain conditions, the constant
+pH method can yield equivalent results as the reaction ensemble :cite:`landsgesell16b`. However, it
+treats the chemical potential of :math:`H^{+}` ions and their actual
+number in the simulation box as independent variables, which can lead to
+serious artifacts. 
+The constant pH method can be used within the reaction ensemble module by
+initializing the reactions with the standard commands of the reaction ensemble. 
+
+The dissociation constant, which is the input of the constant pH method, is the equilibrium
+constant :math:`K_c` for the following reaction:
+
+.. math::
+
+   \mathrm{HA \rightleftharpoons\ H^+ + A^- } \,,
+
+For an example of how to setup
+a Constant pH simulation, see the file in the testsuite directory. 
+For a description of the available methods see :mod:`espressomd.reaction_ensemble`:
+
+.. _Grand canonical ensemble simulation using the Reaction Ensemble:
+
+Grand canonical ensemble simulation using the Reaction Ensemble
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a special case, all stoichiometric coefficients on one side of the chemical
+reaction can be set to zero.  Such reaction creates particles *ex nihilo*, and 
+is equivalent to exchange with a reservoir. Then the simulation in the reaction ensemble becomes equivalent with the
+grandcanonical simulation. Formally, this can be expressed by the reaction
+
+.. math::
+ 
+    \mathrm{\emptyset \rightleftharpoons\ \nu_A A  }  \,, 
+
+where, if :math:`\nu_A=1`, the reaction constant :math:`\Gamma` defines the chemical potential of species A.
+However, if :math:`\nu_A\neq 1`, the statistics of the reaction ensemble becomes
+equivalent to the grandcanonical only in the limit of large average number of species A in the box.
+If the reaction contains more than one product, then the reaction constant
+:math:`\Gamma` defines only the sum of their chemical potentials but not the
+chemical potential of each product alone.
+
+.. Since the Reaction Ensemble acceptance transition probability can be
+.. derived from the grand canonical acceptance transition probability we
+.. can use the reaction ensemble to implement grand canonical simulation
+.. moves. This is done via adding reactions that only have reactants (for the
+.. deletion of particles) or only have products (for the creation of
+.. particles). There exists a one to one mapping of the expressions in the
+.. grand canonical transition probabilities and the expressions in the
+.. reaction ensemble transition probabilities.
```

Comment:
Does this belong to the commented section below?

Reply:
it is not published in a paper. but I think we can uncomment it.